src/cmd/cli/command/commands.go

```diff
@@ -193,8 +193,8 @@ func SetupCommands(version string) {
 	RootCmd.AddCommand(getVersionCmd)
 
 	// Config Command (was: secrets)
-	configSetCmd.Flags().BoolP("name", "n", false, "name of the config (backwards compat)")
 	configSetCmd.Flags().BoolP("env", "e", false, "set the config from an environment variable")
```

Comment:
As a user, I don't think it's clear how to actually use this flag, and a quick glance at the docker docs doesn't shed any light on what they expect either.

I assume I should write something like `-e FOO` to expose the current value in my shell, but how do I set more than one var?

Should I use `-e FOO -e BAR`, or `-e "FOO, BAR"`

These questions occur to me as well:
* Can I pass a value explicitly with something like `-e FOO=bar`?
  * If so, can I mix and match literal values and values from the shell? `-e FOO=bar,BAZ`
* Aren't config values exposed to my container through the environment?
* If a value is set in as a defang config value, but it's also in my compose file, and in a shell flag, what are the precedence rules?

Could we add an example invocation with multiple values to the help text? More information would probably be more appropriate in a docs page. It would be cool to include a short link inline to that page though.

Reply:
> Should I use -e FOO -e BAR, or -e "FOO, BAR"

Like Docker: `-e FOO -e BAR`

> Can I pass a value explicitly with something like -e FOO=bar?

Like Docker: no

> Aren't config values exposed to my container through the environment?

Yes, but not from you shell.

> If a value is set in as a defang config value, but it's also in my compose file, and in a shell flag, what are the precedence rules?

Config has precedence.

Reply:
> > Can I pass a value explicitly with something like -e FOO=bar?
>
> Like Docker: no

I'm not sure I follow this. With Docker, you can pass a value like that. I just ran `docker compose run --rm -e TEST=stuff web` in a sample and it worked fine?

Reply:
ugh yah that's different though. I agree this is confusing. `docker compose` always exposes your shell env to the compose loader, so they get used always. `docker compose -e …` can be used to override env vars. `defang` does not expose your shell env (by design), and I was thinking how to make it explicit, the way `docker run` does (not compose!) using `-e`. But now I see `docker compose` also has a `-e` and it does something else, so never mind.

Reply:
> Config has precedence

This actually feels counter-intuitive to me. I sort of have a mental model that the value which is most volatile should take precedent. In this case, the config is sort of the _least_ volatile, the flag is the most. If I sorted them from most precedent to least, I think I would expect the following:

1. command line flag
2. compose file env
3. dockerfile env
4. config

Reply:
Yeah, that would be my instinct as well ^

Reply:
This Raph guy here disagrees with you @raphaeltm : https://github.com/DefangLabs/defang/issues/449

Reply:
> This Raph guy here disagrees with you @raphaeltm : #449

😂 It's funny, when I was chatting with @jordanstephens later I was like "Well actually... it is kinda like the config value is being passed in and overriding... so maybe that does make more sense?" 

I think my conclusion is that there isn't an obvious order.

Reply:
RE: config has precedence. I agree with @jordanstephens's hierarchy of precedence; to me it seems analogous with scoping in many programming languages. local variables overrides file level variables overrides globals. cli over compose/dockerfile over config. If I'm making an explicit effort to set a value in a compose/dockerfile I would not expect it to be ignored for a config value.